### PR TITLE
Support IFile.Copy

### PR DESCRIPTION
**Pull request by timbussmann**
implements the Copy(...) methods on the InMemoryFile and unifies the implementation on InMemoryFile and InMemoryFileInfo.

### CODE REVIEW

File: source/Appccelerate.IO.Facts/Access/InMemory/InMemoryFileSystemExtensionsFacts.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        }
+
+        [Fact]
+        public void EnsuresDirectoryForFilePathExists()
+        {
+            var directoryPath = @"C:\some\directory\path";
+            this.fileSystem.CreateDirectory(directoryPath);
+
+            this.fileSystem.EnsureParentDirectoryExists(Path.Combine(directoryPath, "test.txt"));
+        }
```
Comment (ursenzler) [inline source/Appccelerate.IO.Facts/Access/InMemory/InMemoryFileSystemExtensionsFacts.cs]: missing assertion?

Comment (ursenzler) [inline source/Appccelerate.IO.Facts/Access/InMemory/InMemoryFileSystemExtensionsFacts.cs]: missing assertion?

Comment (timbussmann) [inline source/Appccelerate.IO.Facts/Access/InMemory/InMemoryFileSystemExtensionsFacts.cs]: good catch, I'll fix that

**Comment by ursenzler** [comment]
Thanks!

**Pull request merged**